integration-tests/ahm/src/tests.rs

```diff
@@ -43,17 +44,41 @@ use xcm_emulator::ConvertLocation;
 
 use super::mock::*;
 
+type RcChecks = (
+	pallet_rc_migrator::preimage::PreimageChunkMigrator<Polkadot>,
+	pallet_rc_migrator::indices::IndicesMigrator<Polkadot>,
+	// other pallets go here
+);
+
+type AhChecks = (
+	pallet_rc_migrator::preimage::PreimageChunkMigrator<AssetHub>,
+	pallet_rc_migrator::indices::IndicesMigrator<AssetHub>,
+	// other pallets go here
+);
+
 #[tokio::test(flavor = "multi_thread", worker_threads = 2)]
-async fn preimage_pallet_migration() {
-	let Some((rc, ah)) = load_externalities().await else { return };
-	type RcPayload = <pallet_rc_migrator::preimage::PreimageChunkMigrator<Polkadot> as RcPalletMigrationChecks>::RcPayload;
-	let (dmp_messages, rc_payload) =
-		rc_migrate::<pallet_rc_migrator::preimage::PreimageChunkMigrator<Polkadot>>(rc);
-	// TODO: fix failing post-migration checks
-	ah_migrate::<
-		pallet_rc_migrator::preimage::PreimageChunkMigrator<Polkadot>,
-		pallet_ah_migrator::preimage::PreimageMigrationCheck<AssetHub>,
-	>(ah, rc_payload, dmp_messages);
+async fn pallet_migration_works() {
+	let Some((mut rc, mut ah)) = load_externalities().await else { return };
+
+	// Pre-checks on the Relay
+	let rc_pre_payload = rc.execute_with(RcChecks::pre_check);
+
+	// Pre-checks on the Asset Hub
+	let ah_pre_payload = ah.execute_with(|| AhChecks::pre_check(rc_pre_payload.clone()));
```

Comment:
how about not running checks if `START_STAGE` is provided? we consider it as a debug no checks run?